src/org/labkey/test/components/domain/BaseDomainDesigner.java

```diff
@@ -0,0 +1,100 @@
+package org.labkey.test.components.domain;
+
+import org.labkey.test.BootstrapLocators;
+import org.labkey.test.Locator;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+/**
+ * Automates the LabKey ui component defined in: packages/components/src/components/domainproperties/BaseDomainDesigner.tsx
+ * Defines standard save, cancel, and error handling
+ */
+public abstract class BaseDomainDesigner<EC extends BaseDomainDesigner.ElementCache> extends WebDriverComponent<EC>
+{
+    private final WebElement el;
+    private final WebDriver driver;
+
+    protected BaseDomainDesigner(WebDriver driver)
+    {
+        this.driver = driver;
+        el = Locator.id("app").findElement(this.driver); // Full page component
+    }
+
+    @Override
+    public WebElement getComponentElement()
+    {
+        return el;
+    }
+
+    @Override
+    public WebDriver getDriver()
+    {
+        return driver;
+    }
+
+    public boolean isCancelButtonEnabled()
+    {
+        return elementCache().cancelButton.isEnabled();
+    }
+
+    public Object clickCancel()
+    {
+        clickCancel(true);
+        return null;
+    }
+
+    protected final void clickCancel(boolean expectPageLoad)
+    {
+        if (expectPageLoad)
+        {
+            getWrapper().doAndWaitForPageToLoad(() -> elementCache().cancelButton.click());
+        }
+        else
+        {
+            elementCache().cancelButton.click();
+        }
+    }
+
+    public boolean isSaveButtonEnabled()
+    {
+        return elementCache().saveButton.isEnabled();
+    }
+
+    public Object clickSave()
```

Comment:
Type `Object` is so that subclasses can return a specific page (see `QueryMetadataEditor` or `SampleSetDesignPage`). The `clickSave(boolean)` and `clickCancel(boolean)` methods were to allow subclasses to specify whether a page load is expected. I wasn't happy with the solution and have deleted them in favor of having subclasses interact directly with the buttons in the element cache.